src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -242,14 +282,71 @@ public override void ExecuteCmdlet()
-                size: Size);
+            ResourceConfig<VirtualMachine> virtualMachine = null;
+
+            if (image != null)
+            {
+                virtualMachine = resourceGroup.CreateVirtualMachineConfig(
+                    name: Name,
+                    networkInterface: networkInterface,
+                    isWindows: isWindows,
+                    adminUsername: Credential.UserName,
+                    adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
+                    image: image,
+                    size: Size);
+            }
+            else
+            {
+                var storageClient =
+                        AzureSession.Instance.ClientFactory.CreateArmClient<StorageManagementClient>(DefaultProfile.DefaultContext,
+                            AzureEnvironment.Endpoint.ResourceManager);
+                var st1 = storageClient.StorageAccounts.Create(ResourceGroupName, Name, new StorageAccountCreateParameters
+                {
+                    AccountType = AccountType.PremiumLRS,
+                    Location = Location
+                });
+                var filePath = new FileInfo(SessionState.Path.GetUnresolvedProviderPathFromPSPath(DiskFile));
+                using (var vds = new VirtualDiskStream(filePath.FullName))
+                {
+                    if (vds.DiskType == DiskType.Fixed)
+                    {
+                        long divisor = Convert.ToInt64(Math.Pow(2, 9));
+                        long rem = 0;
+                        Math.DivRem(filePath.Length, divisor, out rem);
+                        if (rem != 0)
+                        {
+                            throw new ArgumentOutOfRangeException("filePath", string.Format("Given vhd file '{0}' is a corrupted fixed vhd", filePath));
+                        }
+                    }
+                }
+                var storageAccount = storageClient.StorageAccounts.GetProperties(ResourceGroupName, Name);
+                BlobUri destinationUri = null;
+                BlobUri.TryParseUri(new Uri(string.Format("{0}{1}/{2}{3}", storageAccount.PrimaryEndpoints.Blob, Name.ToLower(), Name.ToLower(), ".vhd")), out destinationUri);
+                if (destinationUri == null || destinationUri.Uri == null)
+                {
+                    throw new ArgumentNullException("destinationUri");
+                }
+                var storageCredentialsFactory = new StorageCredentialsFactory(this.ResourceGroupName, storageClient, DefaultContext.Subscription);
+                var parameters = new UploadParameters(destinationUri, null, filePath, true, 2)
+                {
+                    Cmdlet = this,
+                    BlobObjectFactory = new CloudPageBlobObjectFactory(storageCredentialsFactory, TimeSpan.FromMinutes(1))
+                };
+                if (!string.Equals(Environment.GetEnvironmentVariable("AZURE_TEST_MODE"), "playback", StringComparison.OrdinalIgnoreCase))
```

Comment:
@markcowl @hyonholee 
Is it OK to do so?

Reply:
hmm..  I don't think this should be in dev code.

Reply:
This is to increase testability & I think I somehow saw similar code somewhere.